android/src/main/kotlin/io/customer/customer_io/CustomerIoPlugin.kt

```diff
@@ -1,35 +1,111 @@
-    if (call.method == "getPlatformVersion") {
-      result.success("Android ${android.os.Build.VERSION.RELEASE}")
-    } else {
-      result.notImplemented()
+class CustomerIoPlugin : FlutterPlugin, MethodCallHandler {
+    /// The MethodChannel that will the communication between Flutter and native Android
+    ///
+    /// This local reference serves to register the plugin with the Flutter Engine and unregister it
+    /// when the Flutter Engine is detached from the Activity
+    private lateinit var channel: MethodChannel
+    private lateinit var context: Context
+
+    private val logger: Logger
+        get() = CustomerIOShared.instance().diGraph.logger
+
+    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        context = flutterPluginBinding.applicationContext
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "customer_io")
+        channel.setMethodCallHandler(this)
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        when (call.method) {
+            "getPlatformVersion" -> {
+                result.success("Android-${android.os.Build.VERSION.RELEASE}")
+            }
+            "initialize" -> {
+                initialize(call, result)
+            }
+            else -> {
+                result.notImplemented()
+            }
+        }
+    }
+
+
+    private fun initialize(call: MethodCall, result: Result) {
+        try {
+            val application: Application = context.applicationContext as Application
+            val configData = call.arguments as? Map<String, Any> ?: emptyMap()
+            val siteId = configData.getString(Keys.Environment.SITE_ID)
+            val apiKey = configData.getString(Keys.Environment.API_KEY)
+            val region = configData.getProperty<String>(
+                Keys.Environment.REGION
+            )?.takeIfNotBlank().toRegion()
+            
+            CustomerIO.Builder(
+                siteId = siteId,
+                apiKey = apiKey,
+                region = region,
+                appContext = application,
+            ).apply {
+                setClient(client = getUserAgentClient(packageConfig = configData))
+                setupConfig(configData)
+            }.build()
+            logger.info("Customer.io instance initialized successfully")
+            result.success(true)
+        } catch (e: Exception) {
+            logger.error("Failed to initialize Customer.io instance from app, ${e.message}")
+            result.error("FlutterSegmentException", e.localizedMessage, null);
+        }
+    }
+
+    private fun getUserAgentClient(packageConfig: Map<String, Any?>?): Client {
+        val sourceSDKVersion = packageConfig?.getProperty<String>(
+            Keys.PackageConfig.SOURCE_SDK_VERSION
+        )?.takeIfNotBlank() ?: packageConfig?.getProperty<String>(
+            Keys.PackageConfig.SOURCE_SDK_VERSION_COMPAT
```

Comment:
I am not sure what `Keys` is. 

Is this code saying that Flutter customers will set in their apps the value for `SOURCE_SDK_VERSION`? Or are we able to set that value ourselves?